lib/src/cached_network_image_provider.dart

```diff
@@ -54,15 +71,27 @@ class CachedNetworkImageProvider
   }
 
   Future<ui.Codec> _loadAsync(CachedNetworkImageProvider key) async {
-    var mngr = cacheManager ?? DefaultCacheManager();
-    var file = await mngr.getSingleFile(url, headers: headers);
+    final manager = useScaleCacheManager
+        ? ScaledImageCacheManager()
+        : (cacheManager ?? DefaultCacheManager());
+    final modifiedUrl = _transformedUrl(manager, url);
+    var file = await manager.getSingleFile(modifiedUrl, headers: headers);
```

Comment:
Aren't you downloading a different url here?

Reply:
They don't need to be understood by the server these are just locally interpreted params, the only thing is they shouldn't collide with server specified params for which there is provision in the cache config to specify custom query keys

Reply:
@creativepsyco am I missing anything?